services/tas/tas-tests.service.js

```diff
@@ -0,0 +1,110 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import {
+  testResultQueryParamSchema,
+  renderTestResultBadge,
+} from '../test-results.js'
+
+const commonAttrs = {
+  namedParams: {
+    provider: 'github',
+    org: 'test-at-scale',
+    repo: 'badge-demo',
+  },
+  queryParams: {
+    passed_label: 'passed',
+    failed_label: 'failed',
+    skipped_label: 'skipped',
+    compact_message: null,
+  },
+}
+
+const schema = Joi.object({
+  badge: Joi.object({
+    passed: Joi.number().required(),
+    failed: Joi.number().required(),
+    skipped: Joi.number().required(),
+    total_tests: Joi.number().required(),
+    status: Joi.string().required(),
```

Comment:
Not necessary to make any changes here, but sharing for awareness/future reference that we do have some schema helpers as well (`nonNegativeInteger` from `validators.js` would have been useful)

Reply:
thanks for the nit. I've updated this in the latest changes.